Sources/UInt128.swift

```diff
@@ -193,44 +193,44 @@ extension UInt128 : FixedWidthInteger {
     
     // MARK: Instance Methods
     
-    public func addingReportingOverflow(_ rhs: UInt128) -> (partialValue: UInt128, overflow: ArithmeticOverflow) {
-        var resultOverflow = ArithmeticOverflow.none
+    public func addingReportingOverflow(_ rhs: UInt128) -> (partialValue: UInt128, overflow: Bool) {
+        var resultOverflow = false
         let (lowerBits, lowerOverflow) = self.value.lowerBits.addingReportingOverflow(rhs.value.lowerBits)
         var (upperBits, upperOverflow) = self.value.upperBits.addingReportingOverflow(rhs.value.upperBits)
         
         // If the lower bits overflowed, we need to add 1 to upper bits.
-        if lowerOverflow == .overflow {
+        if lowerOverflow {
             (upperBits, resultOverflow) = upperBits.addingReportingOverflow(1)
         }
         
-        let hasOverflowed = (upperOverflow == .overflow) || (resultOverflow == .overflow)
+        let hasOverflowed = (upperOverflow == true) || (resultOverflow == true)
         
         return (partialValue: UInt128(upperBits: upperBits, lowerBits: lowerBits),
-                overflow: ArithmeticOverflow(hasOverflowed))
+                overflow: hasOverflowed)
     }
     
-    public func subtractingReportingOverflow(_ rhs: UInt128) -> (partialValue: UInt128, overflow: ArithmeticOverflow) {
-        var resultOverflow = ArithmeticOverflow.none
+    public func subtractingReportingOverflow(_ rhs: UInt128) -> (partialValue: UInt128, overflow: Bool) {
+        var resultOverflow = false
         let (lowerBits, lowerOverflow) = self.value.lowerBits.subtractingReportingOverflow(rhs.value.lowerBits)
         var (upperBits, upperOverflow) = self.value.upperBits.subtractingReportingOverflow(rhs.value.upperBits)
         
         // If the lower bits overflowed, we need to subtract (borrow) 1 from the upper bits.
-        if lowerOverflow == .overflow {
+        if lowerOverflow {
             (upperBits, resultOverflow) = upperBits.subtractingReportingOverflow(1)
         }
         
-        let hasOverflowed = (upperOverflow == .overflow) || (resultOverflow == .overflow)
+        let hasOverflowed = (upperOverflow == true) || (resultOverflow == true)
         
         return (partialValue: UInt128(upperBits: upperBits, lowerBits: lowerBits),
-                overflow: ArithmeticOverflow(hasOverflowed))
+                overflow: hasOverflowed)
```

Comment:
Would you mind removing line 222 and changing `overflow: hasOverflowed` to `overflow: upperOverflow || resultOverflow`?